### PR TITLE
Add endpoints to generic parser for JSON format

### PR DESCRIPTION
**Pull request by damiencarol**
Add support for endpoints for Generic parser (JSON)

```JSON
{
    "findings": [
        {
            "title": "test title with endpoints as dict",
            "description": "Some very long description with\n\n some UTF-8 chars à qu'il est beau",
            "severity": "Medium",
            "mitigation": "Some mitigation",
            "date": "2021-01-06",
            "cve": "CVE-2020-36234",
            "cwe": 261,
            "cvssv3": "CVSS:3.1/AV:N/AC:L/PR:H/UI:R/S:C/C:L/I:L/A:N",
            "file_path": "src/first.cpp",
            "line": 13,
            "endpoints": [
                {
                    "host": "exemple.com"
                }
            ]
        },
        {
            "title": "test title with endpoints as strings",
            "description": "Some very long description with\n\n some UTF-8 chars à qu'il est beau2",
            "severity": "Critical",
            "mitigation": "Some mitigation",
            "date": "2021-01-06",
            "cve": "CVE-2020-36235",
            "cwe": 287,
            "cvssv3": "CVSS:3.1/AV:N/AC:L/PR:H/UI:R/S:C/C:L/I:L/A:N",
            "file_path": "src/two.cpp",
            "line": 135,
            "endpoints": [
                "http://urlfiltering.paloaltonetworks.com/test-command-and-control",
                "https://urlfiltering.paloaltonetworks.com:2345/test-pest"
            ]
        },
        {
            "title": "test title",
            "description": "Some very long description with\n\n some UTF-8 chars à qu'il est beau2",
            "severity": "Critical",
            "mitigation": "Some mitigation",
            "date": "2021-01-06",
            "cve": "CVE-2020-36236",
            "cwe": 287,
            "cvssv3": "CVSS:3.1/AV:N/AC:L/PR:H/UI:R/S:C/C:L/I:L/A:N",
            "file_path": "src/threeeeeeeeee.cpp",
            "line": 1353
        }
    ]
}
```

### CODE REVIEW

File: docs/content/en/integrations/import.md [hunk 1]
```diff
@@ -228,7 +228,78 @@ Import Findings from XML file format.
...
+ - Impact
+ - References
+ - Verified ('TRUE', 'FALSE')
+ - FalsePositive
+ - CVE
+ - CweId
+ - CVSSV3
+ - Url
+
+Exemple of JSON format:
```
Comment (valentijnscholten) [inline docs/content/en/integrations/import.md:250]:
```suggestion
Example of JSON format:
```

File: dojo/tools/generic/parser.py [hunk 2]
```diff
@@ -30,12 +30,27 @@ def get_findings_json(self, filename, test, active=None, verified=None):
         data = json.load(filename)
         findings = list()
         for item in data['findings']:
+            # remove endpoints of the dictionnary
+            unsaved_endpoints = None
+            if "endpoints" in item:
+                unsaved_endpoints = item["endpoints"]
+                del item["endpoints"]
+
```
Comment (valentijnscholten) [inline dojo/tools/generic/parser.py:38]: also add clean() as per kibliks PR?

**Review by valentijnscholten — CHANGES_REQUESTED** [review]

File: dojo/tools/generic/parser.py (hunk 2 above)
Comment (damiencarol) [reply dojo/tools/generic/parser.py:38]: I think it's better to add clean in the (re)importer
People that implement parsers will forget it.
Also do it in (re)importer will make things at the same level as the unit tests. I mean it's the object that use parsers and results of `get_findings()` that checks endpoints.

Comment (valentijnscholten) [reply dojo/tools/generic/parser.py:38]: sorry I meant in the testcase, but it's already there I see.

**Review by valentijnscholten — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Pull request merged**